pubsub-mapped-api/src/main/java/com/google/pubsub/clients/consumer/ConsumerConfig.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.pubsub.clients.consumer;
+
+import com.google.pubsub.kafkastubs.common.config.AbstractConfig;
+import com.google.pubsub.kafkastubs.common.config.ConfigDef;
+import com.google.pubsub.kafkastubs.common.config.ConfigDef.Importance;
+import com.google.pubsub.kafkastubs.common.config.ConfigDef.Type;
+import com.google.pubsub.kafkastubs.common.serialization.Deserializer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * The consumer configuration keys
+ */
+class ConsumerConfig extends AbstractConfig {
+  private static final ConfigDef CONFIG;
+
+    /*
+     * NOTE: DO NOT CHANGE EITHER CONFIG STRINGS OR THEIR JAVA VARIABLE NAMES AS
+     * THESE ARE PART OF THE PUBLIC API AND CHANGE WILL BREAK USER CODE.
```

Comment:
the ConsumerConfig file is almost entirely based on Kafka implementation. 

https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

as this is a public class there, i am not sure if users are not using it. could you please take a look? we can rewrite it from scratch our way if this is better